daemon/k8s_watcher.go

```diff
@@ -257,40 +255,47 @@ func (d *Daemon) EnableK8sWatcher(reSyncPeriod time.Duration) error {
 	)
 	go ingressController.Run(wait.NeverStop)
 
-	ciliumNetworkPolicyHandler := cache.ResourceEventHandlerFuncs{
-		AddFunc:    d.addCiliumNetworkPolicy,
-		UpdateFunc: d.updateCiliumNetworkPolicy,
-		DeleteFunc: d.deleteCiliumNetworkPolicy,
-	}
-
-	var ciliumRulesController cache.Controller
-	ciliumRulesStore, ciliumRulesController = cache.NewInformer(
-		cnpClient.NewListWatch(),
-		&cilium_crd.CiliumNetworkPolicy{},
-		reSyncPeriod,
-		ciliumNetworkPolicyHandler,
-	)
-
-	stopCiliumRulesController := make(chan struct{})
-	go ciliumRulesController.Run(stopCiliumRulesController)
+	si := informer.NewSharedInformerFactory(ciliumNPClient, reSyncPeriod)
+
+	createHandlerStore := func(si informer.SharedInformerFactory, version int) (cnpStore cache.Store) {
```

Comment:
Why is this an inline func?

Reply:
I had restructured the code differently and before I needed the inline func. I've remove it.